### DISCUSSION TITLE
Calculate, export and import the Inverse Polefigure (IPDF)

### DISCUSSION THREAD

**Discussion by tarekiraki** [mtex-toolbox/mtex#1402] [Ask Anything]
Dear Sir or Madam,

I read the examples to (1) calculate an ODF, (2) calculate the polefigure and (3) export the polefigure on the basis of orientation data. I also read the example to reconstruct the ODF from polefigure data [PoleFigure2ODFAmbiguity](https://mtex-toolbox.github.io/PoleFigure2ODFAmbiguity.html). The source code for this pipeline is listed below.

I would like to test my pipeline with the Inverse Polefigure (IPDF). I saw in this example [ODFInversePoleFigure](https://mtex-toolbox.github.io/ODFInversePoleFigure.html) how to plot the IPDF via the function e.g. `plotIPDF(odf,[xvector,zvector],'complete','upper')` or `plotIPDF(odf,[xvector,zvector],'antipodal')`.

My question is, if there is a way to calculate, export and import the IPDF like the 'polefigure-way'?

Thanks for help.

My Pipeline:
```
% calculate ODF
ebsd = EBSD.load(
	filename_ori_data, CS, 'interface', 'generic', 'ColumnNames', 
	{'x' 'y' 'phi1' 'Phi' 'phi2' 'weights'}, 'Bunge');
oris = ebsd.orientations;
odf = calcDensity(oris, 'halfwidth', 3*degree, 'weights', ebsd.weights);

% calculate Pole Figure
pf_odf = calcPoleFigure(odf, h, 'upper');

% export Pole Figure
export(pf_odf, folder_pf, 3*degree)

% import Pole Figure
pf_import = loadPoleFigure(
	fnames, cs, h, 'interface','generic', 'ColumnNames', 
	{'Polar Angle' 'Azimuth Angle' 'Intensity'}, 'RADIANS');

% calulate ODF based on Pole Figure data
odf_recon = calcODF(pf_import,'silent');

% export odf_recon
export(odf_recon, filename_odf_export)
```

**Comment by kilir** [comment]
Hi,
maybe something like this, but you need to keep track of the crystalSymmetry.

```
%example
mtexdata titanium
oris = ebsd.orientations;
odf = calcDensity(oris,'exact');
% calculate Pole Figure
ipdf = calcPDF(odf,[],xvector);
g = equispacedS2Grid
S.value = eval(ipdf,g)
export(g,'test.out',S)
```
Cheers,
Rüdiger

**Comment by tarekiraki** [comment]
Hi Rüdiger, 

thanks for your answer. With your advice I am able to export the IPDF. With the equispacedS2Grid default settings, the result is an 6600x4 matrix with the columns x, y, z, value. 

Now I have two questions:

(1) Is there a description how to interpret the output like for the export-polefigure function (https://mtex-toolbox.github.io/PoleFigure.export.html)? I noticed that z goes from 1 to -1 and that y goes to 0 at irregular intervals. Is there a way to visualize this output file via matlab/mtex or python?

(2) Is there a way to import the IPDF on the basis of this output like the load-polefigure function? My first attempt does not work:
`pf_import = loadPoleFigure(ipf_file_name, CS, h, 'interface','generic','ColumnNames', {'x' 'y' 'z', 'value'});`

Cheers Tarek
